tools/build_script_generator/scons/site_tools/info.py

```diff
@@ -78,39 +85,42 @@ def git_info_defines(env, with_status=False):
 			for line in s:
 				if len(line.strip()) > 0:
 					c = line.strip()[0]
-					f[c] = f[c] + 1
-			defines['MODM_GIT_STATUS']    = "1"
-			defines['MODM_GIT_MODIFIED']  = f['M']
-			defines['MODM_GIT_ADDED']     = f['A']
-			defines['MODM_GIT_DELETED']   = f['D']
-			defines['MODM_GIT_RENAMED']   = f['R']
-			defines['MODM_GIT_COPIED']    = f['C']
-			defines['MODM_GIT_UPDATED_NOT_MERGED'] = f['U']
-			defines['MODM_GIT_UNTRACKED'] = f['?']
-		# Format and escape all defines
-		defines = {k:"\\\"{}\\\"".format(str(v).replace("(", "\\(").replace(")", "\\)")) for k, v in defines.items() }
+					f[c] += 1
+			defines['MODM_GIT_STATUS'] = "1"
+			subs['MODM_GIT_MODIFIED']  = f['M']
+			subs['MODM_GIT_ADDED']     = f['A']
+			subs['MODM_GIT_DELETED']   = f['D']
+			subs['MODM_GIT_RENAMED']   = f['R']
+			subs['MODM_GIT_COPIED']    = f['C']
+			subs['MODM_GIT_UPDATED_NOT_MERGED'] = f['U']
+			subs['MODM_GIT_UNTRACKED'] = f['?']
+
 	except subprocess.CalledProcessError as e:
 		env.Error('failed to run git command: %s' % e)
+		return
 
-	# We're not creating a header file for these, because all files are supposed to be checked
-	# in and checking in the header file with the git info utterly defeats the point of this.
-	# Instead we're adding these defines via the command line.
 	env.AppendUnique(CPPDEFINES=defines)
+	target = join(env['BASEPATH'], 'src', 'info_git.h')
+	subs = {"type": "git", "defines": {k:"\"{}\"".format(v) for k, v in subs.items()}}
+	return env.Jinja2Template(target=target, source=TEMPLATE_SOURCE, substitutions=subs)
+
 
 def build_info_defines(env):
 	cwd = env.Dir('#').abspath
 	defines = {}
-	defines['MODM_BUILD_PROJECT_NAME'] = env.get('CONFIG_PROJECT_NAME', 'Unknown')
-	defines['MODM_BUILD_MACHINE'] = platform.node()
-	defines['MODM_BUILD_USER'] = getpass.getuser()
+	subs = {}
+	defines['MODM_BUILD_INFO'] = "1"
+	subs['MODM_BUILD_PROJECT_NAME'] = env.get('CONFIG_PROJECT_NAME', 'Unknown')
+	subs['MODM_BUILD_MACHINE'] = platform.node()
+	subs['MODM_BUILD_USER'] = getpass.getuser()
 	# Generate OS String
 	if platform.system() == 'Linux':
 		os = " ".join(platform.linux_distribution())
```

Comment:
`platform.linux_distribution()` is deprecated since Python 3.5 and will be removed in Python 3.8. It's not deprecated in python 2 though. On arch Linux SCons uses Python 2 by default for which `platform.linux_distribution()` returns `('', '', '')` for me. If we decide to keep it, we should probably add a check for empty string and just put `Linux` into MODM_BUILD_OS if no distro name is available.

Reply:
No wonder, this code was taken from xpcc. I’ll fix this, it’ll only be an issue later.

Reply:
So… the alternative is to `pip install distro` and use that. But I don't want to have to install another package just for this. I'll probably just revert to `platform.platform()`, which isn't the same, but close enough I guess. We'll fix it for real if someone needs it.